pkg/plugins/envvars/envvars.go

```diff
@@ -100,6 +100,8 @@ func (s *Service) Get(ctx context.Context, p *plugins.Plugin) []string {
 }
 
 // GetConfigMap returns a map of configuration that should be passed in a plugin request.
+//
+//nolint:gocyclo
```

Comment:
i'm not happy about adding this, but the problem is, we reached a limit of `IFs` in this function by me adding the `if s.cfg.UserFacingDefaultError != "" ` part. and there's no way to nicely remove that one. we would have to remove other `IFs` from the function, and that would be complicated, and i'm not even sure if we want to do it.
still, if it needs to be done, just tell me and i can try to create a separate PR that reduces the complexity first. 👍